cpp/example_code/s3/put_object_acl.cpp

```diff
@@ -0,0 +1,242 @@
+    if (type == "Amazon customer by email")
+        return Aws::S3::Model::Type::AmazonCustomerByEmail;
+    if (type == "Canonical user")
+        return Aws::S3::Model::Type::CanonicalUser;
+    if (type == "Group")
+        return Aws::S3::Model::Type::Group;
+    return Aws::S3::Model::Type::NOT_SET;
+}
+
+// snippet-end:[s3.cpp.put_object_acl.code]
+
+/*
+ *
+ * main function
+ *
+ * Prerequisites: An existing bucket that contains the object you want to get
+ * information about the ACL for.
+ *
+ * TODO(User) items: Set the following variables
+ * - bucketName: The name of the bucket. For example,
+ *   "DOC-EXAMPLE-BUCKET".
+ * - objectKey: The name of the object to get ACL information
+ *   about. For example, "my-file.txt".
+ * - owner_id: Canonical ID for owner.
+ * - Select Which form of grantee you want to specify, and update the corresponding data
+ * - Uncomment additional parameters if necessary.
+*
+ */
+
+#ifndef TESTING_BUILD
+
+int main() {
+    Aws::SDKOptions options;
+    Aws::InitAPI(options);
+    {
+        //TODO(User): Change bucket_name to the name of a bucket in your account.
+        const Aws::String bucket_name = "<Enter a bucket name>";
+        //TODO(User): Name of object in bucket.
+        const Aws::String object_name = "<Enter object name>";
+
+
+        //TODO(User): Set owner_id to your canonical id.  It is your bucket so you are the ACL owner.
+        //See https://docs.aws.amazon.com/AmazonS3/latest/userguide/finding-canonical-user-id.html for more information.
+        const Aws::String owner_id =
+                "<Enter owner canonical id>";
+
+        // Set the ACL's grantee information.
+        const Aws::String grantee_permission = "READ"; //Give the grantee Read permissions.
+
+        //TODO(User): Select which form of grantee (grantee_type) you want, then you must update the data
+        // corresponding to that selected type.
+
+        // If the grantee is by canonical user, then you must specify either the user's ID or 
+        // grantee_display_name:
+        const Aws::String grantee_type = "Canonical user";
+        const Aws::String grantee_id =
+                "<Enter owner canonical id>";
+        // const Aws::String grantee_display_name = "janedoe";
+
+        // If the grantee is by Amazon customer by email, then you must specify the email 
+        // address:
+        // const Aws::String grantee_type = "Amazon customer by email";
+        // const Aws::String grantee_email_address = "janedoe@example.com";
+
+        // If the grantee is by group, then you must specify the predefined group URI:
+        //const Aws::String grantee_type = "Group";
+        //// This example uses one of Amazon S3 predefined groups: Authenticated Users group (Access permission to this group allows any AWS account to access the resource. )
+        //// See https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html#specifying-grantee for more information.
```

Comment:
//// For more information, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html#specifying-grantee.